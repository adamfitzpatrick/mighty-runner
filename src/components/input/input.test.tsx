import { h } from 'preact'
import { shallow } from 'preact-render-spy'
import Input from '.'

describe('Input component', () => {
  it('should render with the provided label & value', () => {
    const input = shallow(<Input label='label' value='value' onChange={() => { /* no-op */ }} />)
    expect(input).toMatchSnapshot()
  })

  it('should call the provided onChange handler when the input is changed', () => {
    const handler = jest.fn()
    const input = shallow(<Input label='label' value='value' onChange={handler} />)
    input.find('input').simulate('change')
    expect(handler).toHaveBeenCalled()
  })
})
