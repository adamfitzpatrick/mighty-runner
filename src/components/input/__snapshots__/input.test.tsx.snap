// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `"<label><span>label</span><input data-testid=\\"label.input.component\\" type=\\"text\\" value=\\"value\\"></label>"`;

exports[`Input component should render without a blur prop 1`] = `"<label><span>label</span><input data-testid=\\"label.input.component\\" type=\\"text\\" value=\\"value\\"></label>"`;
